src/opera/util/input_validation.py

```diff
@@ -307,23 +307,30 @@ def validate_disp_inputs(runconfig, logger, name):
     dyn_anc_file_group = runconfig.sas_config['dynamic_ancillary_file_group']
     static_anc_file_group = runconfig.sas_config['static_ancillary_file_group']
 
-    check_input_list(input_file_group['cslc_file_list'], logger, name,
-                     valid_extensions=('.h5',), check_zero_size=True)
+    if runconfig.pge_name == 'DISP_S1_PGE':
+        check_input_list(input_file_group['cslc_file_list'], logger, name,
+                         valid_extensions=('.h5',), check_zero_size=True)
 
-    cslc_burst_id_set = get_cslc_input_burst_id_set(
-        runconfig.sas_config['input_file_group']['cslc_file_list'],
-        logger, name
-    )
+        cslc_burst_id_set = get_cslc_input_burst_id_set(
+            runconfig.sas_config['input_file_group']['cslc_file_list'],
+            logger, name
+        )
 
-    if ('static_layers_files' in dyn_anc_file_group and
-            isinstance(dyn_anc_file_group['static_layers_files'], list) and
-            len(dyn_anc_file_group['static_layers_files']) > 0):
-        check_input_list(dyn_anc_file_group['static_layers_files'], logger, name,
+        if ('static_layers_files' in dyn_anc_file_group and
+                isinstance(dyn_anc_file_group['static_layers_files'], list) and
+                len(dyn_anc_file_group['static_layers_files']) > 0):
+            check_input_list(dyn_anc_file_group['static_layers_files'], logger, name,
+                             valid_extensions=('.h5',), check_zero_size=True)
+            check_disp_s1_ancillary_burst_ids(cslc_burst_id_set,
+                                              dyn_anc_file_group['static_layers_files'],
+                                              logger,
+                                              name)
+    else:
+        check_input_list(input_file_group['gslc_file_list'], logger, name,
                          valid_extensions=('.h5',), check_zero_size=True)
-        check_disp_s1_ancillary_burst_ids(cslc_burst_id_set,
-                                          dyn_anc_file_group['static_layers_files'],
-                                          logger,
-                                          name)
+
+        # TODO: Is there a NI equivalent to the burst set - ancillaries validation of DISP-S1?
+        #  I don't seem to see a burst ID in the input filenames
```

Comment:
Had a look at the sample GSLC and GUNW products we got from the interface delivery, and while I did not see anything that could be correlated in the filenames, we could at the very least ensure the runconfig specifies the same number of each file. We'll need to double-check with ADT that this is a true invariant though.

Reply:
Should we do that in this PR or check with ADT first?

Reply:
Lets go ahead and add it now